### ISSUE TITLE
A test BaseBufferTest throws a different type of exception on JVM.

### ISSUE THREAD

**Issue by nicolasstucki** [scala-js/scala-js#2190]
``` scala
if (!executingInJVM) // throws BufferOverflowException on JVM
  expectThrows(classOf[ReadOnlyBufferException], buf.put(Array[ElementType](6, 7, 12)))
```

**Comment by sjrd** [comment]
It turns out the JVM sometimes throws `ReadOnlyBufferException`, and sometimes `BufferOverflowException`. The Javadoc does not favor one or the other in case we try to put an array too large into a buffer that's read-only anyway.

So the test is too restrictive here. The implementation is correct.

**Comment by gzm0** [comment]
Fixed in 46435b537d9720576862d5cf74263f73b0b9f6ad
